pkg/alertmanager/multitenant.go

```diff
@@ -1262,16 +1262,42 @@ func safeTemplateFilepath(dir, templateName string) (string, error) {
 		return "", err
 	}
 
+	// If actualPath is same as containerDir, it's likely that actualPath was empty, or just ".".
+	if containerDir == actualPath {
+		return "", fmt.Errorf("invalid template name %q", templateName)
+	}
+
 	// Ensure the actual path of the template is within the expected directory.
 	// This check is a counter-measure to make sure the tenant is not trying to
 	// escape its own directory on disk.
-	if !strings.HasPrefix(actualPath, containerDir) {
+	if !isFilePathInsideDirectory(containerDir, actualPath) {
 		return "", fmt.Errorf("invalid template name %q: the template filepath is escaping the per-tenant local directory", templateName)
 	}
 
 	return actualPath, nil
 }
 
+// Returns true if file described by filePath is inside given directory. (Doesn't check for symbolic links)
+// Directory argument should not end with separator, ie. it must be filepath.Clean-ed.
+func isFilePathInsideDirectory(absDir string, absFilePath string) bool {
```

Comment:
nit: can you make the requirements of this function a little more clear? I.e. "(Doesn't check for symbolic links so the paths provided must already be absolute paths via filepath.Abs())" or similar?